CIP-????/README.md

```diff
@@ -0,0 +1,530 @@
+combine the pieces will be discussed later in the spec. 
+
+#### Block
+
+The cddl for a block is currently:
+
+```cddl
+block =
+  [ header
+  , transaction_bodies         : [* transaction_body]
+  , transaction_witness_sets   : [* transaction_witness_set]
+  , auxiliary_data_set         : {* transaction_index => auxiliary_data }
+  , invalid_transactions       : [* transaction_index ]
+  ]; Valid blocks must also satisfy the following two constraints:
+   ; 1) the length of transaction_bodies and transaction_witness_sets
+   ;    must be the same
+   ; 2) every transaction_index must be strictly smaller than the
+   ;    length of transaction_bodies
+```
+
+Just like with the transaction cddl, the block's transaction bodies and the transaction witness sets
+will be merged, and the constraints must be updated accordingly:
+
+```cddl
+; This is re-posted here for reference.
+transaction_piece =
+  { 0: transaction_body
+  , 1: transaction_witness_set ; The witness set for this transaction body.
+  }
+
+; A specific transaction in the block is a specific batch of transaction pieces. 
+block_transaction = [* transaction_piece]
+
+block =
+  [ header
+  , block_transactions   : [* block_transaction] ; This is new
+  , auxiliary_data_set   : {* transaction_index => auxiliary_data }
+  , invalid_transactions : [* transaction_index ]
+  ]; Valid blocks must also satisfy the following constraint:
+   ; 1) every transaction_index must be strictly smaller than the
+   ;    length of block_transactions
+```
+
+### Transaction Size Limit
+
+A transaction broken into pieces should still have approximately the same total size as the whole
+transaction. There are a few things to note:
+
+- If two transaction pieces require the same witness (ie, both pieces require a signature from
+Alice), this witness would now effectively appear twice.
+- Certain items can appear in multiple pieces (eg, the same input can be spent/referenced, tokens can
+be minted/burned that would otherwise cancel out, etc).
+
+Despite the above, it is probably fine to leave the maximum transaction size as is, at least to
+start. The transaction fee would naturally incentivize users/aggregators to minimize the amount of
+duplication in a transaction.
+
+### Block Size Limit
+
+Due to the possible duplicate items in each transaction and the fact that a block is made up of
+multiple transactions, the issues in the Transaction Size Limit section are slightly exacerbated.
+However, it should still be fine to keep the current block size limit, at least initially. Just like
+with transactions, the transaction fee would naturally incentivize users to minimize duplications.
+
+### Smart Contract Execution
+
+While the cryptographic keys only need to sign the relevant transaction pieces, the smart contracts
+would be executed against the full transaction. This requirement enables the current smart contract
```

Comment:
> Without a good reason for this, it's probably better to properly reflect this structure in `TxInfo`.

Personally, whichever option minimizes added complexity is my preferred choice. If we preserve the transaction as you are suggesting, the smart contract interface would need a pretty large change and DApps would need to be re-architected. Perhaps keeping the structure in `TxInfo` would be better in the long run, but I'm not sure. I do not know what changes would be required on the ledger/plutus side to actually preserve the structure in `TxInfo`.

@vlasin If we did preserve the structure, I believe this would address your [suggestion](https://github.com/cardano-foundation/CIPs/pull/873#issuecomment-2270494230).